xdsl/parser.py

```diff
@@ -356,9 +356,11 @@ def parse_optional_char(self,
 
     def parse_char(self, char: str, skip_white_space: bool = True) -> bool:
         assert (len(char) == 1)
+        current_char = self.get_char()
```

Comment:
helped me with debugging, might help someone else in the future

Reply:
My personal opinion is that the code is fairly self-explanatory, I'm not sure what the dosctring would say. Happy to include one if you have a suggestion!